### ISSUE TITLE
item-sliding stops working if the item-option is open and the list changes dynamically 

### ISSUE THREAD

**Issue by pelupotter** [ionic-team/ionic#15599]
# Bug Report

item-sliding stops working if the item-option is open and the list changes dynamically (Observable async)

**Ionic Info**
```
Ionic:

   ionic (Ionic CLI)          : 4.1.2 (/usr/lib/node_modules/ionic)
   Ionic Framework            : @ionic/angular 4.0.0-beta.7
   @angular-devkit/core       : 0.7.5
   @angular-devkit/schematics : 0.7.5
   @angular/cli               : 6.1.5
   @ionic/ng-toolkit          : 1.0.8
   @ionic/schematics-angular  : 1.0.6

Cordova:

   cordova (Cordova CLI) : 8.0.0
   Cordova Platforms     : none
   Cordova Plugins       : no whitelisted plugins (0 plugins total)

System:

   NodeJS : v8.11.3 (/usr/bin/node)
   npm    : 6.4.0
   OS     : Linux 4.15
```

**Describe the Bug**
item-sliding stops working if the item-option is open and the list changes dynamically (Observable async)

**Steps to Reproduce**
1. open an item-sliding and leave it open until you change the list that contains it (dynamically)

**Expected Behavior**
keep running normally

**Additional Context**

tested in the Chrome browser

**Comment by sean-perkins** [comment]
If you force close all the slide items before/after the list changes, it should allow the item-option to continue to behave correctly. Ran into a similar scenario in my app, and calling `slideItem.close()` fixed future behavior.

**Comment by eweren** [comment]
Yes, having this issue still in beta.17

**Comment by brandyscarney** [comment]
This was solved in the `4.0.2` release (latest is `4.1.1`) please update! If you're still seeing this issue please let me know but I cannot reproduce it in latest, thanks!
